### DISCUSSION TITLE
Exception Handler $internalDontReport should not apply to console

### DISCUSSION THREAD

**Discussion by amcsi** [laravel/framework#42975] [Ideas]
Today I encountered a problem where an exception was not reported to Sentry.

The thing is that it was an HttpException that was thrown during a scheduled command.

Although I understand that the list of classes in [$internalDontReport](https://github.com/laravel/framework/blob/6e0fb37ca7d432e95d1386bdb1dab56b9b36ffd7/src/Illuminate/Foundation/Exceptions/Handler.php#L94-L106) are important when invoking Laravel via the web (be it traditionally rendered or through the API) in order to avoid common, expected errors from being spammed to the log.

But when running Laravel code in the console, I believe that entire list should be empty, because even those types of exceptions are all unexpected, and thus should be reported to any logs by default.

Of course one would need to be mindful that Octane is technically run in the console, but acts web-like, so if we were to make sure the internal report list is empty for console, it should not be empty for Octane.

**Comment by MattApril** [comment]
Interesting that there isn't more discussion around this, or any solutions offered. I can't really think of a scenario where I want to ignore an exception during an async job.

A quick and dirty solution is:
```
->withExceptions(function (Exceptions $exceptions) {
        # Prevent Laravel's default ignored exceptions when running in console
        if (App::runningInConsole() && !App::runningUnitTests()) {
            $ref = new ReflectionClass($exceptions->handler);
            $prop = $ref->getProperty('internalDontReport');
            $prop->setAccessible(true);
            $internalDontReport = $prop->getValue($exceptions->handler);
            $exceptions->stopIgnoring($internalDontReport);
        }
})
```
However, this does not address Octane usage, as the OP mentioned.

EDIT: If you use Octane you can expand the above using \Laravel\Octane\Facades\Octane::isRunning()
